packages/flutter/lib/src/services/text_input.dart

```diff
@@ -276,6 +293,8 @@ class TextInputConnection {
 
 TextInputAction _toTextInputAction(String action) {
   switch (action) {
+    case 'TextInputAction.newline':
+      return TextInputAction.newline;
     case 'TextInputAction.done':
       return TextInputAction.done;
```

Comment:
you should keep the switch and the enum definition in the same order unless the order matters